src/run_program.rs

```diff
@@ -484,18 +514,20 @@ impl<'a, D: Dialect> RunProgramContext<'a, D> {
                 None => break,
             };
             cost += match op {
-                Operation::Apply => augment_cost_errors(
-                    self.apply_op(cost, effective_max_cost - cost),
-                    max_cost_ptr,
-                )?,
+                Operation::Apply => {
+                    let apply_op_res = self.apply_op(cost, effective_max_cost - cost);
+                    augment_cost_errors(apply_op_res, max_cost_ptr)?
+                }
```

Comment:
It seems like this little `diff` is the exact same code, yes?

Reply:
ya changed that back.